src/main/java/mate/jdbc/dao/CarDaoImpl.java

```diff
@@ -0,0 +1,261 @@
+            ResultSet resultSet = statement.executeQuery();
+            if (resultSet.next()) {
+                car = getCar(resultSet);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't get car by id  " + id, e);
+        }
+        if (car != null) {
+            car.setDrivers(getDrivers(id));
+        }
+        return Optional.ofNullable(car);
+    }
+
+    @Override
+    public List<Car> getAll() throws SQLException {
+        Car newCar = null;
+        List<Car> cars = new ArrayList<>();
+        List<Driver> drivers = new ArrayList<>();
+        String query = "SELECT cars.id, "
+                + "cars.model, cars.manufacturer_id, "
+                + " manufacturers.name, manufacturers.country "
+                + "FROM cars "
+                + "JOIN manufacturers "
+                + "ON manufacturers.id = manufacturer_id "
+                + "ORDER BY cars.id;";
+        System.out.println(query);
+        ResultSet resultSet = null;
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            resultSet = statement.executeQuery();
+            while (resultSet.next()) {
+                newCar = getCar(resultSet);
+                cars.add(newCar);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't get all cars", e);
+        }
+        for (Car car : cars) {
+            car.setDrivers(getDrivers(car.getId()));
+        }
+        return cars;
+    }
+
+    @Override
+    public Car update(Car car) {
+        String query = "UPDATE cars SET model = ?, manufacturer_id = ? "
+                + "WHERE cars.id = ?;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setString(1, car.getModel());
+            statement.setLong(2,car.getManufacturer().getId());
+            statement.setLong(3,car.getId());
+            statement.executeUpdate();
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't udate car by id " + car.getId(), e);
+        }
+        for (Driver tempDriver: car.getDrivers()) {
+            removeDriverFromCar(tempDriver,car);
+        }
+        for (Driver tempDriver: car.getDrivers()) {
+            addDriverToCar(tempDriver,car);
+        }
+        return car;
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        String query = "UPDATE cars SET is_deleted = true WHERE id = ?";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setLong(1, id);
+            int numberOfDeletedRows = statement.executeUpdate();
+            return numberOfDeletedRows != 0;
```

Comment:
```suggestion
            return statement.executeUpdate() != 0;
```
dont create redundant variables